### PR TITLE
Make `git` branch resolution in script more more tolerant

### PR DESCRIPTION
**Pull request by JackPGreen**
The script queries if the branch name starts with `release`, using the `git symbolic-ref` command.

[While testing the `client-compatibility-suites`](https://github.com/hazelcast/client-compatibility-suites/actions/runs/9908837208/job/27376177353), it was noticed that for detached branches, [this command throws an error (`fatal: ref HEAD is not a symbolic ref`) and stops the script](https://stackoverflow.com/a/45028375).

### CODE REVIEW

File: hz.ps1 [hunk 1]
```diff
@@ -1541,7 +1541,7 @@ function hz-build {
         $options.constants = $options.constants.Replace(";", "%3B") # escape ';'
     }
 
-    $branchName = git symbolic-ref --short HEAD
+    $branchName = git name-rev --name-only HEAD
```
Comment (emreyigit) [inline hz.ps1:1544]: It doesn't produce same result. Instead, could you add `-q` to suppress the error?

Comment (JackPGreen) [reply hz.ps1:1544]: > Instead, could you add `-q` to suppress the error?
No, because it still sets the error level / exit code to `1`.

Comment (emreyigit) [reply hz.ps1:1544]: Well, this won't work. `name-rev` doesn't give the full refs path which includes the _release_ directory.

Comment (JackPGreen) [reply hz.ps1]: > Well, this won't work. `name-rev` doesn't give the full refs path which includes the _release_ directory.

Ok, good point.

b72e3a9de98017e484299a293e8dc35318b37413

**Pull request merged**